packages/node/src/lib/token-manager.ts

```diff
@@ -0,0 +1,290 @@
+    this.tokenEmitter.on('access_token', (event) => {
+      if ('token' in event) {
+        this.accessToken = event.token
+      }
+    })
+    this.controller = new AbortController()
+    this.signal = this.controller.signal
+    this.signal.addEventListener('abort', () => {
+      clearTimeout(this.pollerTimer)
+    })
+    this.retryCount = 0
+  }
+
+  async pollerLoop() {
+    let timeUntilRefreshInMs = 25
+    let response: HTTPResponse
+
+    try {
+      response = await this.requestAccessToken()
+    } catch (err) {
+      // Error without a status code - likely networking, retry
+      this.retryCount++
+      this.lastError = err
+
+      if (this.retryCount % this.maxRetries == 0) {
+        this.tokenEmitter.emit('access_token', { error: this.lastError })
+      }
+
+      this.pollerTimer = setTimeout(
+        this.pollerLoop.bind(this),
+        backoff({
+          attempt: this.retryCount,
+          minTimeout: 25,
+          maxTimeout: 1000,
+        })
+      ).unref()
+      return
+    }
+
+    if (response.headers !== undefined && response.headers.Date != undefined) {
+      const serverTime = Date.parse(response.headers.Date)
+      const skew = Date.now() - serverTime
+      if (this.clockSkewInSeconds == 0) {
+        this.clockSkewInSeconds = skew
+      } else {
+        this.clockSkewInSeconds = (this.clockSkewInSeconds + skew) / 2
+      }
+    }
+
+    // Handle status codes!
+    if (response.status === 200) {
+      let body: any
+      if (typeof response.text != 'function') {
+        this.tokenEmitter.emit('access_token', {
+          error: new Error(
+            'HTTPClient does not implement response.text method'
+          ),
+        })
+        clearTimeout(this.pollerTimer)
+        return
+      }
+      try {
+        body = await response.text() //body?.getReader().read() // TODO: Replace with actual method to get body - needs discussion since different HTTP clients expose this differently (buffers, streams, strings, objects)
+      } catch (err) {
+        // Errors reading the body (not parsing) are likely networking issues, we can retry
+        this.retryCount++
+        this.lastError = err
+        timeUntilRefreshInMs = backoff({
+          attempt: this.retryCount,
+          minTimeout: 25,
+          maxTimeout: 1000,
+        })
+        this.pollerTimer = setTimeout(
+          this.pollerLoop.bind(this),
+          timeUntilRefreshInMs
+        ).unref()
+      }
+      let token: AccessToken
+      try {
+        const parsedBody = JSON.parse(body)
+        // TODO: validate JSON
+        token = parsedBody
+        this.tokenEmitter.emit('access_token', { token })
+
+        // Reset our failure count
+        this.retryCount = 0
+
+        // Refresh the token after half the expiry time passes
+        if (token !== undefined && token.expires_in !== undefined) {
+          timeUntilRefreshInMs = Math.floor((token.expires_in / 2) * 1000)
```

Comment:
```suggestion
          timeUntilRefreshInMs = (token.expires_in / 2) * 1000
```
The `Math.floor()` isn't necessary since we're multiplying by 1000 anyways